frame/contracts/src/exec.rs

```diff
@@ -537,13 +563,16 @@ where
 
 	fn instantiate(
 		&mut self,
-		code_hash: &CodeHash<T>,
+		code_hash: CodeHash<T>,
 		endowment: BalanceOf<T>,
 		gas_meter: &mut GasMeter<T>,
 		input_data: Vec<u8>,
 		salt: &[u8],
 	) -> Result<(AccountIdOf<T>, ExecReturnValue), ExecError> {
-		self.ctx.instantiate(endowment, gas_meter, code_hash, input_data, salt)
+		let executable = E::from_storage(code_hash, &self.ctx.config.schedule)?;
```

Comment:
sidenote: is there any way to do `instantiate_with_code` from a contract ? is it planned ? it feels like a legit usecase to me.

Reply:
No it is not possible. We could think about adding this in the future. I am not aware of someone requesting this feature.

Reply:
ah ok, also I see know that it is difficult because code refcount is cached by executable.